packages/payload/src/uploads/safeFetch.ts

```diff
@@ -0,0 +1,71 @@
+import type { Dispatcher } from 'undici'
+
+import ipaddr from 'ipaddr.js'
+import { Agent, fetch as undiciFetch } from 'undici'
+
+const isSafeIp = (ip: string) => {
+  try {
+    if (!ip) {
+      return false
+    }
+
+    if (!ipaddr.isValid(ip)) {
+      return false
+    }
+
+    const parsedIpAddress = ipaddr.parse(ip)
+    const range = parsedIpAddress.range()
+    if (range !== 'unicast') {
+      return false // Private IP Range
+    }
+  } catch (ignore) {
+    return false
+  }
+  return true
+}
+
+const ssrfFilterInterceptor: Dispatcher.DispatcherComposeInterceptor = (dispatch) => {
+  return (opts, handler) => {
+    const url = new URL(opts.origin?.toString() + opts.path)
+    if (!isSafeIp(url.hostname)) {
```

Comment:
Hostname is not always a valid IP address, as per [MDN](https://developer.mozilla.org/en-US/docs/Web/API/URL/hostname) it can be also a domain. Right now, checked on localhost and vercel deployment when editing a previously uploaded file, the hostname is being a domain name.